### PR TITLE
typescript/general: Add more general comments for typescript

### PR DESCRIPTION
**Pull request by SleeplessByte**

### CODE REVIEW

File: analyzer-comments/typescript/general/error_captured_no_source.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# error captured no source
+
+Expected source file `%{expected}`, found: `%{available}`.
+
+The test file `import`s from the expected source file. Ensure the file exists,
+and is correctly named.
```
Comment (iHiD) [inline analyzer-comments/typescript/general/error_captured_no_source.md:6]: Merge onto the same line?

Comment (SleeplessByte) [inline analyzer-comments/typescript/general/error_captured_no_source.md:6]:
```suggestion
The test file `import`s from the expected source file. 
Ensure the file exists, and is correctly named.
```

File: analyzer-comments/typescript/general/function_not_optimal.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# function not optimal
+
+It looks like `%{function}` is not optimal.
+This analyzer is a Work In Progress and can't tell you exactly why it thinks something is not optimal.
+Feel free to ignore this feedback.
+If you wish to attempt to resolve it, the advice is as follows: this function is expected to be as simple as possible, without declaring any extra variables.
```
Comment (ErikSchierboom) [inline analyzer-comments/typescript/general/function_not_optimal.md:6]: When would this comment be posted? It sounds a bit odd to post "hey there is something wrong," but then not be pointed to what actually _is_ wrong.

**Review by ErikSchierboom — APPROVED** [review]

File: analyzer-comments/typescript/general/function_not_optimal.md (hunk 2 above)
Comment (SleeplessByte) [reply analyzer-comments/typescript/general/function_not_optimal.md:6]: It's when you do really weird stuff. I think it's generated once or twice per 1000 solutions and usually it's when people do something like:

```typescript
function twoFer(...args: unknown[]) {
  if (!args[0] || typeof args[0] !== 'string') {
    throw new Error('boom')
  }
  
  const realArg: string = args[0] as string
  // [...]
}
```

so yeah.

**Pull request merged**